src/org/sosy_lab/java_smt/solvers/SolverLess/SMTLIB2Formula.java

```diff
@@ -0,0 +1,380 @@
+// Copyright (C) 2007-2016  Dirk Beyer
+// SPDX-FileCopyrightText: 2025 2020 Dirk Beyer <https://www.sosy-lab.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.sosy_lab.java_smt.solvers.SolverLess;
+
+import java.util.Objects;
+import org.sosy_lab.java_smt.api.ArrayFormula;
+import org.sosy_lab.java_smt.api.BitvectorFormula;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.FloatingPointFormula;
+import org.sosy_lab.java_smt.api.Formula;
+import org.sosy_lab.java_smt.api.FormulaType;
+import org.sosy_lab.java_smt.api.NumeralFormula;
+import org.sosy_lab.java_smt.api.NumeralFormula.IntegerFormula;
+import org.sosy_lab.java_smt.api.NumeralFormula.RationalFormula;
+
+@SuppressWarnings({"StringCaseLocaleUsage", "rawtypes", "Immutable"})
+public class SMTLIB2Formula
```

Comment:
This class is severely overloaded, hard to maintain, massively increases overhead and should be refactored into the appropriate formula-classes!